test/StdCheats.t.sol

```diff
@@ -335,41 +335,52 @@ contract StdCheatsTest is Test {
         return number;
     }
 
-    function testAssumeNoPrecompiles(address addr) external {
-        assumeNoPrecompiles(addr, getChain("optimism_goerli").chainId);
+    function testAssumeAddressIsNot(address addr) external {
+        // skip over Payable and NonPayable enums
+        for (uint8 i = 2; i < uint8(type(AddressType).max); i++) {
+            assumeAddressIsNot(AddressType(i), addr);
+        }
+        assertTrue(addr != address(0));
+        assertTrue(addr < address(1) || addr > address(9));
         assertTrue(
-            addr < address(1) || (addr > address(9) && addr < address(0x4200000000000000000000000000000000000000))
-                || addr > address(0x4200000000000000000000000000000000000800)
+            addr != 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D || addr != 0x000000000000000000636F6e736F6c652e6c6f67
         );
     }
 
-    function _assumePayable(address addr) public {
-        assumePayable(addr);
+    function testAssumeNoPayable() external {
+        // all should pass since these addresses are not payable
+
+        // VM address
+        assumeNoPayable(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+
+        // Console address
+        assumeNoPayable(0x000000000000000000636F6e736F6c652e6c6f67);
+
+        // Create2Deployer
+        assumeNoPayable(0x4e59b44847b379578588920cA78FbF26c0B4956C);
     }
 
-    function testAssumePayable() external {
-        // We deploy a mock version so we can properly test the revert.
-        StdCheatsMock stdCheatsMock = new StdCheatsMock();
+    function testAssumeNoNonPayable() external {
         // all should revert since these addresses are not payable
 
         // VM address
         vm.expectRevert();
-        stdCheatsMock.exposed_assumePayable(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+        assumeNoNonPayable(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
```

Comment:
This will break in forge v1 since JUMPs to internal functions are not supported for `vm.expectRevert` (the fact that they ever were supported was an accident, and they are not well supported—only the first assertion in this test will actually run, then execution silently halts, so it's a bad footgun 😅). So let's make sure to keep the original approach of making sure `expectRevert` is only every used with external calls (hence the `stdCheatsMock.exposed_assumePayable` method). More info in the foundry changelog and https://github.com/foundry-rs/foundry/pull/4945

Let's also add some success cases here that are epected to not revert, e.g. EOAs and a mock payable contract

Reply:
Acknowledged and done.